frontend/src/api/apiManager.js

```diff
@@ -0,0 +1,134 @@
+        const clientKey = `${provider}_${model}`;
+        
+        // Return existing client if available
+        if (this.clients[clientKey]) {
+            return this.clients[clientKey];
+        }
+
+        let client;
+        switch (provider) {
+            case PROVIDERS.GEMINI:
+                client = new GeminiClient(apiKey, model);
+                break;
+            case PROVIDERS.CLAUDE:
+                client = new ClaudeClient(apiKey, model);
+                break;
+            case PROVIDERS.OPENAI:
+                client = new OpenAIClient(apiKey, model);
+                break;
+            case PROVIDERS.GROK:
+                client = new GrokClient(apiKey, model);
+                break;
+            case PROVIDERS.OPENROUTER:
+                client = new OpenRouterClient(apiKey, model);
+                break;
+            case PROVIDERS.CUSTOM_OPENAI:
+                client = new CustomOpenAIClient(apiKey, model, baseUrl);
+                break;
+            default:
+                throw new Error(`Unsupported API provider: ${provider}`);
+        }
+
+        this.clients[clientKey] = client;
+        return client;
+    }
+
+    /**
+     * Generate content using the specified provider
+     * @param {string} provider - The API provider
+     * @param {string} apiKey - The API key
+     * @param {string} model - The model to use
+     * @param {Object} params - Parameters for content generation
+     * @param {string} [baseUrl] - Custom base URL (for custom_openai only)
+     * @returns {Promise<Object>} The generated content response
+     */
+    async generateContent(provider, apiKey, model, params, baseUrl = null) {
+        // Resolve actual API key if encrypted
+        const actualApiKey = await this.resolveApiKey(provider, apiKey);
+        const client = this.getClient(provider, actualApiKey, model, baseUrl);
+        return await client.generateContent(params);
+    }
+
+    /**
+     * Generate a character profile using the specified provider
+     * @param {string} provider - The API provider
+     * @param {string} apiKey - The API key
+     * @param {string} model - The model to use
+     * @param {Object} params - Parameters for profile generation
+     * @param {string} [baseUrl] - Custom base URL (for custom_openai only)
+     * @returns {Promise<Object>} The generated profile response
+     */
+    async generateProfile(provider, apiKey, model, params, baseUrl = null) {
+        // Resolve actual API key if encrypted
+        const actualApiKey = await this.resolveApiKey(provider, apiKey);
+        const client = this.getClient(provider, actualApiKey, model, baseUrl);
+        return await client.generateProfile(params);
+    }
+
+    /**
+     * Clear cached clients (useful when API keys change)
+     */
+    clearClients() {
+        this.clients = {};
+    }
+
+    /**
+     * Get list of supported providers
+     * @returns {Array<string>} List of supported provider names
```

Comment:
Typed more, providing tab completions.
```suggestion
     * @returns {import("../constants/providers.js").SUPPORTED_PROVIDERS} List of supported provider names
```